File: 2주차 과제/[백준] 동전 0.java
import java.util.Scanner;
 
class Main {
	public static void main(String[] args) {
 
		Scanner sc = new Scanner(System.in);

		// 가지고 있는 동전의 종류 N / 동전 금액의 합 K
		int N = sc.nextInt();  
		int K = sc.nextInt();
		
		int[] coin = new int[N];
		
		for(int i = 0; i < N; i++) { // 가지고 있는 동전의 종류 coin 리스트에 삽입
			coin[i] = sc.nextInt();
		}
		
		int count_c = 0;
		// 사용되는 동전의 수 만큼 count
		for(int i = N - 1; i >= 0; i--) {
			if(coin[i] <= K) {
				count_c += (K / coin[i]);
				K = K % coin[i];
			}
		}
		System.out.println(count_c);
	}
 
}

File: 2주차 과제/[백준] 동전 0.py
# 가지고 있는 동전의 종류 N / 동전 금액의 합 K
N, K = map(int,input().split())
coin = []
# 가지고 있는 동전의 종류 coin 리스트에 삽입
for i in range(N):
    coin.append(int(input()))
coin.sort(reverse=True) # 내림차순 정렬

count_c = 0
# 사용되는 동전의 수 만큼 count
for j in coin:
    if K == 0:
        break
    count_c += K // j
    K = K % j

print(count_c)
